task/build-maven-zip/0.1/build-maven-zip.yaml

```diff
@@ -0,0 +1,212 @@
+          sha1sum "$f" | awk '{print $1}'> "$f.sha1"
+          sha256sum "$f" | awk '{print $1}'> "$f.sha256"
+        done <   <(find "$pkgpath/${PKG_ROOT}" -type f ! -name "*.md5" \
+        ! -name "*.sha1" ! -name "*.sha128" ! -name "*.sha256" \
+        ! -name "*.sha512" ! -name "*.asc" -print0)
+        # Bundle the artifacts and checksums together into a zip file
+        cd "$pkgpath"
+        echo "create maven zip to $(workspaces.source.path)/cachi2/output/${BUNDLE_NAME}.zip"
+        zip -rq "${BUNDLE_NAME}.zip" "${PKG_ROOT}"
+        mv "${BUNDLE_NAME}.zip" "/shared/${BUNDLE_NAME}.zip"
+      else
+        echo "No ${PKG_ROOT} dir found, can not generate maven zip!"
+        exit 1
+      fi
+    securityContext:
+      capabilities:
+        add:
+          - SETFCAP
+    workingDir: $(workspaces.source.path)
+  - image: quay.io/konflux-ci/oras:latest@sha256:9d6db5840c70e65fefe041201cc7ffe2d1661bd0582b590b54787213ccfd76e9
+    name: build
+    computeResources:
+      limits:
+        memory: 8Gi
+        cpu: '4'
+      requests:
+        memory: 2Gi
+        cpu: '1'
+    script: |
+      #!/bin/bash
+      set -e
+      if [ -f "/shared/${BUNDLE_NAME}.zip" ]; then
+        mv "/shared/${BUNDLE_NAME}.zip" "./${BUNDLE_NAME}.zip"
+        select-oci-auth "$IMAGE" > auth.json
+        [ -n "$IMAGE_EXPIRES_AFTER" ] && EXPIRE_LABEL=("--annotation" "quay.expires-after=$IMAGE_EXPIRES_AFTER")
+        oras push "$IMAGE" \
+        --registry-config auth.json \
+        "${EXPIRE_LABEL[@]}" \
+        --artifact-type application/vnd.maven+zip "${BUNDLE_NAME}.zip"
+        RESULTING_DIGEST=$(oras resolve --registry-config auth.json "${IMAGE}")
+        echo -n "$IMAGE" | tee "$(results.IMAGE_URL.path)"
+        echo -n "$RESULTING_DIGEST" | tee "$(results.IMAGE_DIGEST.path)"
+      else
+        echo "The maven zip file is not found!"
+        exit 1
+      fi
+
+      # Save the SBOM produced by Cachi2 so it can be merged into the final SBOM later
+      if [ -f "$(workspaces.source.path)/cachi2/output/bom.json" ]; then
+        cp "$(workspaces.source.path)/cachi2/output/bom.json" ./sbom-cachi2.json
+      else
+        echo "The SBOM file for fetched artifacts is not found!"
+        exit 1
+      fi
+    securityContext:
+      capabilities:
+        add:
+          - SETFCAP
+    volumeMounts:
+    - name: trusted-ca
+      mountPath: /mnt/trusted-ca
+      readOnly: true
+    workingDir: $(workspaces.source.path)
+  - name: prepare-sboms
+    image: quay.io/redhat-appstudio/sbom-utility-scripts-image@sha256:53a3041dff341b7fd1765b9cc2c324625d19e804b2eaff10a6e6d9dcdbde3a91
+    computeResources:
+      limits:
+        memory: 512Mi
+        cpu: 200m
+      requests:
+        memory: 256Mi
+        cpu: 100m
+    script: |
+      if [ -f sbom-cachi2.json ]; then
+        mv sbom-cachi2.json sbom-cyclonedx.json
+        echo "Creating sbom-purl.json"
+        python3 /scripts/create_purl_sbom.py
+      fi
```

Comment:
The previous step https://github.com/konflux-ci/build-definitions/pull/1643/files#diff-6199ec974f25eac323954ab607a9160cb1fe2a523abf0e255775588e1e2bda01R136 has already ensured the existence of file `./sbom-cachi2.json`.

Reply:
But, the logic looks ambiguous. In previous step, if the bom.json does not exist, it is an error and will fail the task, then no cachi2-sbom.json is created, then perhaps fail the pipeline, but in this step, the `sbom-cachi2.json` becomes optional for this build task.

Reply:
Ok. I've removed it in new commit.